src/data/random/financial.js

```diff
@@ -1,155 +1,127 @@
-import walk from './walk';
+import _walk from './walk';
+import d3 from 'd3';
 
 export default function() {
-
-    var mu = 0.1,
-        sigma = 0.1,
-        startPrice = 100,
-        startVolume = 100000,
-        startDate = new Date(),
-        stepsPerDay = 50,
-        volumeNoiseFactor = 0.3,
-        filter = function(d) { return true; };
-
-    var calculateOHLC = function(days, prices, volumes) {
-
-        var ohlcv = [],
-            daySteps,
-            currentStep = 0,
-            currentIntraStep = 0;
-
-        while (ohlcv.length < days) {
-            daySteps = prices.slice(currentIntraStep, currentIntraStep + stepsPerDay);
-            ohlcv.push({
-                date: new Date(startDate.getTime()),
-                open: daySteps[0],
-                high: Math.max.apply({}, daySteps),
-                low: Math.min.apply({}, daySteps),
-                close: daySteps[stepsPerDay - 1],
-                volume: volumes[currentStep]
-            });
-            currentIntraStep += stepsPerDay;
-            currentStep += 1;
-            startDate.setUTCDate(startDate.getUTCDate() + 1);
-        }
-        return ohlcv;
+    var startDate = new Date();
+    var startPrice = 100;
+    var granularity = 60 * 60 * 24;
```

Comment:
Yeah I took another look and it seems possible. We would need to pass `step` to get multiples of each interval, and use `interval.offset(date, step)` internally.
E.g. 5 minute intervals:
```javascript
fc.data.random.financial()
  .startDate(new Date(2016, 0, 1))
  .startPrice(100)
  .interval([d3.time.minute, 5]);
```